### PR TITLE
doc/user: Polish release notes for v0.16.0

### PR DESCRIPTION
**Pull request by morsapaes**
Migrating existing release notes under the appropriate release tag. Couldn't single out any other change in the list of PRs worth adding in.

Additionally:

- Added some missing semi-colons that were driving me insane;
- Corrected minor typo in the documentation of the new `list_remove` (in `doc/user/data/sql_funcs.yml`).

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/9993)

### CODE REVIEW

File: doc/user/content/release-notes.md
Comment (morsapaes) [inline doc/user/content/release-notes.md:130]: The `SET` changes were originally two separate notes. Condensed them into a single note to make things less verbose.

Comment (benesch) [reply doc/user/content/release-notes.md:130]: I'd simplify even further and say "Support the [special PostgreSQL syntax `SET NAMES` and `SET SCHEMA`](https://www.postgresql.org/docs/14/sql-set.html#id-1.9.3.173.6)  for setting the `client_encoding` and `search_path` parameters, respectively"? The detail about the semantics for these ops belongs in dedicated documentation on `SET` and `SHOW`... which sadly we don't have yet. Sigh. Regardless I don't think the release notes should be in the business of getting into that level of detail.

Also can we update the link to https://www.postgresql.org/docs/14/sql-set.html#id-1.9.3.173.6? That's the current version of PostgreSQL and links to a slightly more specific section heading.

**Review by benesch — APPROVED** [review]
Woo, we're getting better at getting these right ahead of time! Thanks for handling this week's @morsapaes! I think the one other issue we should mention is https://github.com/MaterializeInc/materialize/issues/9931/https://github.com/MaterializeInc/materialize/pull/9940. Even though it's a pretty obscure bug that Matt found and fixed it's probably worth calling out every crash that gets fixed. Maybe something like this?

> Fix a crash in the optimizer when the branches of a `CASE` statement involved record types whose fields had differing nullability.

File: doc/user/content/release-notes.md
Comment (benesch) [inline doc/user/content/release-notes.md:130]:
```suggestion
- Support the special PostgreSQL syntax [`SET NAMES` and `SET SCHEMA`](https://www.postgresql.org/docs/14/sql-set.html#id-1.9.3.173.6) for setting the `client_encoding` and `search_path` parameters, respectively.
```

File: doc/user/content/release-notes.md
Comment (benesch) [inline doc/user/content/release-notes.md:126]: Pre-existing glitch but this is a broken link. Looks like we don't have a [`numeric`] link yet at the bottom.

**Review by benesch — COMMENTED** [review]
Oh, just noticed two super minor glitches in the rendered preview.

File: doc/user/content/release-notes.md
Comment (morsapaes) [reply doc/user/content/release-notes.md:126]: 🦅

**Comment by morsapaes** [comment]
Thanks for the second pair of eyes (and suggestions), @benesch. 🦅  Pushed the fixes and re-rendered!

**Comment by benesch** [comment]
Rebased and added the relnote for the backported fix to unblock #10035.

**Pull request merged**
